lib/central_mail/utilities.rb

```diff
@@ -48,20 +48,19 @@ def log_submission(uploaded_object, metadata)
 
     def retry_errors(e, uploaded_object)
       if e.code == 'DOC201' && @retries <= RETRIES
-        self.class.perform_at(30.minutes.from_now, uploaded_object.id, { caller: 'DOC201_Retry' }, @retries + 1)
+        self.class.perform_at(30.minutes.from_now, uploaded_object.guid, { caller: 'DOC201_Retry' }, @retries + 1)
       else
         uploaded_object.update(status: 'error', code: e.code, detail: e.detail)
       end
       log_error(e, uploaded_object)
     end
 
     def log_error(e, uploaded_object)
-      uuid = uploaded_object.respond_to?(:guid) ? uploaded_object.guid : uploaded_object.id
```

Comment:
This line is actually what I meant by the `guid -> id` fallback - basically, try to get the uploaded_object's guid, and if it doesn't exist, try for `id` instead. The problem before was that the retry method was forcing everything to use `id` even if the object had `guid` available.

So let's keep this `uuid = ...` line in log_error but also add it to `retry_errors` so we get the most consistent but also widest ranging coverage for ids we can manage :)

Reply:
I see. That's fair. I can tell that these methods were meant for reuse even through they're not being reused right now. That said, I think the functionality of the `UploadProcessor` job and these two `CentralMail::Utilities` methods are a little too intertwined right now. I would eventually like to refactor them so that there's a better separation of concerns. The current implementation assumes that the calling method can handle an `id` or a `guid`, and that's not the case with `UploadProcessor`. In fact, we could avoid spawning a new job entirely if we work within the existing Sidekiq retry framework instead of this special `@retries` variable. But I'll save that work for a future PR.

Reply:
I've made the changes that you requested. Can you take another look?

Thanks!

Reply:
Yeah, the utility is more designed around composition into a job class than being a standalone utility class in its own right. I agree (and it's what AppealsAPI has done) that ignoring the utility's retry in favor of sidekiq's retry functionality makes sense even if it's just from a "principle of least surprise" lens.

Reply:
Great, sounds like we're in agreement. 👍